datameta/api/files.py

```diff
@@ -48,6 +48,60 @@ class FileResponse(FileBase):
     content_uploaded  : bool
     filesize          : Optional[int] = None
 
+
+def delete_staged_file_from_db(file_id, db, auth_user):
+    # Obtain file from database
+    db_file = resource.resource_query_by_id(db, models.File, file_id).one_or_none()
+
+    # Check if file could be found
+    if db_file is None:
+        raise HTTPNotFound(json=None)
+
+    # Check if requesting user has access to the file
+    if db_file.user_id != auth_user.id:
+        raise HTTPForbidden(json=None)
+
+    if db_file.metadatumrecord is not None:
+        raise errors.get_not_modifiable_error()
+
+    user_uuid, file_uuid, storage_uri = db_file.user.uuid, db_file.uuid, db_file.storage_uri
+    
+    # Delete the database record
+    db.delete(db_file)
+
+    return user_uuid, file_uuid, storage_uri
+
+
+@view_config(
+    route_name      = "rpc_delete_files",
+    renderer        = "json",
+    request_method  = "POST",
+    openapi         = True
+)
+def delete_files(request: Request) -> HTTPNoContent:
+    # Check authentication and raise 401 if unavailable
+    auth_user = security.revalidate_user(request)
+
+    db = request.dbsession
+
+    deleted_files = [
+        delete_staged_file_from_db(file_id, db, auth_user)
+        for file_id in set(request.openapi_validated.body["fileIds"])    
+    ]
+
+    # Commit transaction
+    request.tm.commit()
+    request.tm.begin()
+
+    # Delete the files from storage
+    for user_uuid, file_uuid, storage_uri in deleted_files:
+        log.info(f"[DB_FILE][DELETE][user={user_uuid}][file={file_uuid}]")
+        storage.rm(request, storage_uri)
+        log.info(f"[STORAGE][DELETE][user={user_uuid}][file={file_uuid}]")
+
```

Comment:
I think we'll have to loop first for the `[DB_FILE][DELETE]` logs to be generated and then loop again for the storage removal. Otherwise if `storage.rm` fails for any file and raises an exception in the logs it will look as if the remaining files weren't deleted from the database, but they were